lib/cabbage/feature.ex

```diff
@@ -165,7 +165,11 @@ defmodule Cabbage.Feature do
     scenarios = Module.get_attribute(env.module, :scenarios) || []
     steps = Module.get_attribute(env.module, :steps) || []
     tags = Module.get_attribute(env.module, :tags) || []
-    for scenario <- scenarios, test_number <- 1..Kernel.length(scenarios) do
+    test_number = 0
+
+    for scenario <- scenarios do
+      test_number = test_number + 1
```

Comment:
does this work? I thought immutability put a damper in these kind of things... is this just rebinding over and over again?

Reply:
Doesnt seem like it works in the shell

```elixir
iex(2)> i = 0
0
iex(3)> for x <- 1..10 do
...(3)>   i = i + 1
...(3)>   IO.puts i
...(3)> end
1
1
1
1
1
1
1
1
1
1
[:ok, :ok, :ok, :ok, :ok, :ok, :ok, :ok, :ok, :ok]
```

This looks to be what you want:
```elixir
iex(4)> 100..110 |> Enum.with_index() |> Enum.each(fn {item, index} -> IO.puts index end)
0
1
2
3
4
5
6
7
8
9
10
:ok
```
So you should so something like: (untested)
```elixir
Enum.with_index(scenarios) |> Enum.each(fn {scenario, test_number} ->
   # all the rest of the stuff here
end)
```